src/components/number-picker/number-picker.component.less

```diff
@@ -4,12 +4,14 @@ ux-number-picker-inline {
     flex-direction: row;
     height: 34px;
 
-    &.ng-invalid {
-        border-color: @critical;
+    &.ux-number-picker-invalid:not([formControlName]) {
```

Comment:
I think this isn't sufficient because you can also bind a control with `[formControl]`. Possibly `:not(.ng-valid):not(.ng-invalid)`?

Reply:
I cant have <code>:not(.ng-valid)<code> as this does not apply the border when using <code>ngModel</code> as it does not apply the <code>ng-invalid</code> class

Reply:
Do you mave a plunker example? I'm not sure I understand the scenario that you are describing.

Reply:
I think this is the case where they are using `ngModel` to set the value but also using the built in `min` and `max` functionality. ngModel will apply the `ng-valid` class if none of its validation rules fail, but there is still the possibility of the components built in validation failing, so we could have both ng-valid and number-picker-invalid classes present.

Reply:
Yes, that's what I mean ^^

Reply:
So, this is using the current has-error styles but, if you change the value to 20 above the max value, the ng-invalid class doesn't apply and it still says ng-valid. https://plnkr.co/edit/Sri8u348U3JuhdlLu0Xj?p=preview

Reply:
Well you need to use `[min]` and `[max]` for number inputs: https://plnkr.co/edit/pkWyNVVsT2JmO2aTa8rG?p=preview

Reply:
Oh, that was with the old stylesheet, I see the point. What selector will work for this?

Reply:
for that example <code>.ux-number-picker-invalid</code> will be applied

Reply:
`:not(.ng-valid):not(.ng-invalid):not([ngModel])`?

Reply:
Maybe you could do an optional inject of the parent FormGroupDirective, and set `ux-number-picker-valid` only if that is null?

Reply:
I'm not sure if this will work as <code>ngModel</code> doesn't produce a <code>formGroup</code>. What I have currently applies either <code>ng-invalid</code> or the <code>.ux-number-picker-invalid</code>

Reply:
Yes, when `ngModel` is used then we will apply `ux-number-picker-invalid`. When a form group is present, we assume that validation is handled externally.

Reply:
then would <code> &.ux-number-picker-invalid:not(.ng-invalid):not([formControlName])</code> work?

Reply:
The trouble with that is that you can alternatively use `[formControl]="myControl"` to bind a form control.

Reply:
You could add <code>:not[(formControl)]</code> aswell? If not, what way do you want this to be done?

Reply:
I think that wouldn't match because object bindings aren't rendered to the DOM in the same way.

I'd suggest injecting the parent FormGroupDirective into the number picker, and only emitting the `ux-number-picker-validation` class if the directive is null. Does that make sense?

Reply:
Yes, I will do that now.